pkg/asset/fsck_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package asset
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/google/syzkaller/pkg/image"
+	_ "github.com/google/syzkaller/sys/linux/gen"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFsck(t *testing.T) {
+	tests := []struct {
+		name    string
+		image   string
+		isClean bool
+	}{
+		// Extracted from sys/linux/test/syz_mount_image_ext4_0
```

Comment:
Can we use sys/linux/test seeds directly?
Maybe we already have a test that extracts mounted images from all these seeds.
fsck should run successfully on all these seeds. It will also check how it works on more fs types (e.g. check if we give correct fsck arguments).
For corrupted ones may be we could just generate some random garbage. We don't need/want to test that fsck itself works correctly.

Reply:
> Can we use sys/linux/test seeds directly?

Yep, I actually started with this approach, I could restore the logic.

> fsck should run successfully on all these seeds. It will also check how it works on more fs types (e.g. check if we give correct fsck arguments).

One concern I have is that we aren't sure what `fsck`s the user will have installed in the `syz-manager` environment so running `btrfs check` might fail just because the user doesn't have that one command.

> For corrupted ones may be we could just generate some random garbage. We don't need/want to test that fsck itself works correctly.

I tried that too. My naive attempt was to flip all bits of the 10th byte but it wasn't enough to corrupt the ext4 image apparently. That's why I ended up using a fs that is known to be corrupted.

Anyway, in light of the xz attack I agree it's best to avoid introducing unexplained big binary blobs in source code. Corrupted filesystems would be one great place to hide something ahah. So I guess I'll try experimenting with other corruption schemes (maybe zero-ing out the second block or something) and hopefully get better results.